async-nats/src/service/mod.rs

```diff
@@ -0,0 +1,437 @@
+// Copyright 2020-2022 The NATS Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+pub mod error;
+
+use std::{
+    collections::HashMap,
+    fmt::Display,
+    pin::Pin,
+    sync::{Arc, Mutex},
+    time::{Duration, Instant},
+};
+
+use bytes::Bytes;
+use futures::{
+    stream::{self, SelectAll},
+    Future, Stream, StreamExt, TryFutureExt,
+};
+use serde::{Deserialize, Serialize};
+use serde_json::json;
+use time::serde::rfc3339;
+use time::OffsetDateTime;
+use tokio::task::JoinHandle;
+use tracing::debug;
+
+use crate::{client::Publish, Client, Error, HeaderMap, Message, Subscriber};
+
+const SERVICE_API_PREFIX: &str = "$SRV";
+const QUEUE_GROUP: &str = "svc";
```

Comment:
Different than in ADR
```suggestion
const QUEUE_GROUP: &str = "q";
```